### PR TITLE
Enable custom canary reporting to cloudwatch

### PR DESCRIPTION
**Pull request by jsitu777**
- enable canary to push custom metrics to cloudwatch
   - total tests
   - success tests
   - failed tests

### CODE REVIEW

File: test/canary/scripts/push_stats_to_cloudwatch.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        testsuite = tree.find("testsuite")
+        failures = testsuite.attrib["failures"]
+        tests = testsuite.attrib["tests"]
+        successes = int(tests) - int(failures)
+    else:
+        print("f{xml_path} does not exists.")
+        print(os.getcwd())
+        failures = 0
+        successes = 0
+        tests = 1
```
Comment (ananth102) [inline test/canary/scripts/push_stats_to_cloudwatch.py:21]: Why is this 1? Shouldn't it be equal to the amount of tests in the canary suite?

Comment (surajkota) [reply test/canary/scripts/push_stats_to_cloudwatch.py:21]: does it matter 0/1 is still 0?
yes it does give wrong sense of how many tests ran but making it equal to num of tests will not be maintainable

File: test/canary/scripts/push_stats_to_cloudwatch.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        raise
+
+
+def main():
+    readXML_and_publish_metrics_to_cw()
+
+
+if __name__ == "__main__":
+    main()
+
```
Comment (surajkota) [inline test/canary/scripts/push_stats_to_cloudwatch.py:92]: nit: extra lines

File: test/canary/scripts/run_test.sh [hunk 3]
```diff
@@ -63,9 +63,14 @@ function cleanup {
   export PYTHONPATH=.. 
   python service_cleanup.py
 
+  #push to metrics to cloudwatch
```
Comment (surajkota) [inline test/canary/scripts/run_test.sh:66]: lets move this to the top before `clean up resources`

**Review by surajkota — CHANGES_REQUESTED** [review]
few comments, approved with the changes

File: test/canary/scripts/push_stats_to_cloudwatch.py (hunk 1 above)
Comment (jsitu777) [reply test/canary/scripts/push_stats_to_cloudwatch.py:21]: yea it was there for making 0/1=0 so it still sends to cloudwatch if the log was not generated (no tests were run)

**Review by ananth102 — APPROVED** [review]

**Pull request merged**